File: contest/src/Utils/Observable.java
package Utils;

/**
 * Created by Sebi on 15-Nov-16.
 */
public interface Observable<E> {
    /*
    Adds an observer
     */
    void addObserver(Observer<E> o);

    /*
    Removes an observer
     */
    void removeObserver(Observer<E> o);

    /*
    Notifies all observers
     */
    void notifyObservers();

    /*
    Notifies push observers
     */
    void notifyPushObservers(E e);
}

File: contestC#/contest/contest/Utils/Observer/Observable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace contest.Utils.Observer
{
    /*
     * Observable interface 
     */
    public interface Observable
    {
        void notifyObservers();
        void notifyPushObservers(Object e);
        void addObserver(Observer o);
        void removeObserver(Observer o);
    }
}
